src/briefcase/platforms/macOS/__init__.py

```diff
@@ -766,93 +864,234 @@ def package_app(
-                    base_dir=self.binary_path(app).name,
+            else:
+                raise BriefcaseCommandError(
+                    """\
+Your project does not contain a LICENSE file.
+
+Create a file named `LICENSE` in the same directory as your `pyproject.toml`
+with your app's licensing terms.
+"""
                 )
 
-        else:  # Default packaging format is DMG
-            self.logger.info("Building DMG...", prefix=app.app_name)
-
-            with self.input.wait_bar(f"Building {dist_path.name}..."):
-                dmg_settings = {
-                    "files": [os.fsdecode(self.binary_path(app))],
-                    "symlinks": {"Applications": "/Applications"},
-                    "icon_locations": {
-                        f"{app.formal_name}.app": (75, 75),
-                        "Applications": (225, 75),
-                    },
-                    "window_rect": ((600, 600), (350, 150)),
-                    "icon_size": 64,
-                    "text_size": 12,
-                }
+        # pkgbuild's default behavior is to make "relocatable" installs, which means
+        # that if you've ever run the app, the installer will default to updating *that*
+        # version, rather than putting it in the location that the installer specifies.
+        # This means if you've ever used `briefcase run`, that will be the install
+        # location of the "installed" app. To work around this, you have to provide a
+        # plist file - but that requires providing a "root" folder that *only* contains
+        # the products you want to install. So - we need to copy the built app to a
+        # "clean" packaging location.
+        with self.input.wait_bar("Copying app into products folder..."):
+            installed_app_path = installer_path / "root" / self.binary_path(app).name
+            if installed_app_path.exists():
+                self.tools.shutil.rmtree(installed_app_path)
+            self.tools.shutil.copytree(self.binary_path(app), installed_app_path)
+
+        components_plist_path = self.bundle_path(app) / "installer/components.plist"
+
+        with self.input.wait_bar("Writing component manifest..."):
+            with components_plist_path.open("wb") as components_plist:
+                plistlib.dump(
+                    [
+                        {
+                            "BundleHasStrictIdentifier": True,
+                            "BundleIsRelocatable": False,
+                            "BundleIsVersionChecked": True,
+                            "BundleOverwriteAction": "upgrade",
+                            "RootRelativeBundlePath": self.binary_path(app).name,
+                        }
+                    ],
+                    components_plist,
+                )
 
-                try:
-                    icon_filename = self.base_path / f"{app.installer_icon}.icns"
+        # Console apps are installed in /Library/Formal Name, and include the
```

Comment:
Can't console apps also be installed in `~/Library/`?

Reply:
couldn't you soft link `/usr/local/bin/app` to the `~/Library/app` executable in the `postinstall` the same way it does now? i guess come to think of it couldn't you just soft link straight to `/Applications/app`?

just asking because it seems like most of my installed apps are working with `~/Library` via `Application Support/` etc.  instead of the system wide `/Library` and apple is only getting more and more strict about what is allowed to ever touch system wide stuff.

Reply:
That approach won't work if there's more than one user.

Reply:
The Anthropic statement of work says we would install to `/Applications`. That seems to make more sense, because regardless of whether it's a console app or not, it is still an application and not a library.

If this changes, the postinstall script will need to be updated to match.

Reply:
I picked the `/Library` location because macOS will surface any `.app` that exist in `/Applications` or `/Users/xxx/Applications` in the Launchpad, and while you would be able to launch a command line app, it wouldn't display a terminal as a result of being launched, so it would look like a no-op.

Putting console apps in `/Library` seemed broadly consistent with how tools like Xcode handle console-only tooling. It's a location that is clearly a "system" location; but doesn't result in command line apps surfacing in GUI representations.

It's also worth noting that the install location varies depending on the type of app. A GUI app will install into `/Applications`; a console app installs into `/Library`.

Reply:
From in-person discussion, the reason to use /Library is that anything in /Applications will appear in the Launchpad menu in the dock, but a command-line tool can't be launched that way.

It looks like this also applies to subdirectories of /Applications – for example, the official Python installer creates a directory `/Applications/Python 3.x` containing `IDLE.app` and `Python Launcher.app`, both of which appear in the Launchpad.

Reply:
OK, that makes sense, but since it's a change to the spec, let's remember to mention to Anthropic next time we send them an update.